client/src/components/AppointmentCreateForm.js

```diff
@@ -0,0 +1,141 @@
+   ] = useMutation(CREATE_APPOINTMENT, {
+    refetchQueries: [{query: FETCH_APPOINTMENTS}],
+    onCompleted: () => backToLanding()
+  });
+
+  const fromTimeInputProps = {
+    placeholder: "Click to open time picker",
+    id: "pickupFromTime",
+    required: "required"
+  };
+  const toTimeInputProps = {
+    placeholder: "Click to open time picker",
+    id: "pickupToTime",
+    required: "required"
+  };
+  
+  var startTime = new Date(newFromTime);
+  var endTime = new Date(newToTime);
+
+  const handleSubmit = () => {
+    if (startTime>=endTime){
+        return [false, alert("Oops, 'End Time' must happen after 'Start time'! \n Please try again.")]
+    }else{
+      return createAppointment({ 
+        variables: {startTime: newFromTime.format('X'), endTime: newToTime.format('X'), address:newAddress, note:newNote }});
+    }
+  };
+  
+  return (
+    <div>
+    <form onSubmit={e => {
+      e.preventDefault();
+      handleSubmit();
+        }}
+    >
+    <div>
+      <div className="container-fluid m-1">    
+         <label htmlFor="pickupFromTime">Start time
+        <Datetime 
+          inputProps={fromTimeInputProps}
+          dateFormat={true}
+          timeFormat={true}
+          value={newFromTime}
+          onChange={value => setNewFromTime(value)}
+          />
+        </label>
+      </div>
+      
+      <div className="container-fluid m-1">
+        <label htmlFor="pickupToTime">End time
+        <Datetime 
+          inputProps={toTimeInputProps} 
+          dateFormat={true}
+          timeFormat={true}
+          value={newToTime}
+          onChange={value => setNewToTime(value)}
+        />
+        </label>
+      </div>
+
+      <div className="container-fluid m-1">
+          <label htmlFor="pickupLocation">Address
+          <input
+            className="form-control"
+            id="pickupLocation"
+            placeholder="Enter address"
+            value={newAddress}
+            onChange={e => setNewAddress(e.target.value)
+            }
+            required="required"
+          />
+          </label>
+      </div>
+      
+      <div className="container-fluid m-1">
+          <label htmlFor="pickupDetails">Pickup details (Optional)
+          <input
+            className="form-control"
+            id="pickupDetails"
+            placeholder="Further instructions on the pickup location"
+            value={newNote}
+            onChange={e => setNewNote(e.target.value)}
+          />
+          </label>
+      </div>
+      <div className="container-fluid m-1">
+        <button type="submit" className="btn btn-primary mx-1">
+          Save draft
+        </button>
+        <button type="button" onClick={backToLanding} className="btn btn-secondary mx-1">
+          Back
+        </button>
+        </div>
+      </div>
+      </form>   
+      {mutationLoading && <p>Loading...</p>}
+      {mutationError && <p>Error:( Please try again</p>}
```

Comment:
When an error happens, how likely is a retry going to resolve the problem? Don't suggest the users that they should retry if the problem is likely not resolved by retrying.

Reply:
I just changed the sentence to "Error:( We can not accept this appointment." But I think so far we do not need to include this warning since all the functions work well. Only one case I could imagine our app would show this warning is that there were probably some internet issues. When it happened, this warning would show for just a few seconds, and the whole page cashed right after.